src/octoprint/static/js/app/client/logs.js

```diff
@@ -1,30 +0,0 @@
-(function (global, factory) {
```

Comment:
That bit of the js client library should be moved into the plugin as well (and used therein). In order to be backward compatible we'll need a solution as well - someone might already be using `OctoPrintClient.logs.*` and simply moving that over to `OctoPrintClient.plugins.logs.*` without a "redirect" would potentially cause issues in the field.

I'm still a bit unsure how best to do such a redirect, that is currently not supported in some easy way. Maybe something simple as `OctoPrintClient.registerComponent("logs", OctoPrintLogsClient);` next to `OctoPrintClient.registerPluginComponent("logs", OctoPrintLogsClient);` would already do it though (extra nice: deprecation warning on using `OctoPrint.logs.*` instead of `OctoPrint.plugin.logs.*` - maybe by simply providing a trivial wrapper `DeprecatedOctoPrintLogsClient` to `registerComponent` since we only have three methods anyhow?).